test/functional/options/num_options_spec.lua

```diff
@@ -36,6 +37,14 @@ describe(':setlocal', function()
     feed_command('setlocal imsearch=1')
     eq(-1, meths.get_option('imsearch'))
   end)
+
+  it('returns an error if local option does not exist #15587', function()
+    feed_command('setglobal undolevels=1000')
+    feed_command('setlocal undolevels<')
+    eq("Failed to get value for option 'undolevels'",
```

Comment:
IMHO this should also return the sentinel value (not sure why vim choose -123456 for this option and not -2 or something simple but that ship has probably sailed; most global-local options have "sane" sentinels anyway). An unset local value is not an error and thus should not be represented as an error.

Reply:
> An unset local value is not an error and thus should not be represented as an error.

Agreed -- this PR is just a stopgap to address the *immediate* problem of returning an uninitialized value. Fixing the problem more generally still needs to be done. We can just adapt this PR to do that if we don't want to use an intermediate band-aid solution.

Reply:
I just see no reason to implement some code/behaviour we don't want before the code we actually want. Given that we just want the _same_ behaviour as the vimL interfaces I'd assume it is equally small change in `get_option_value_strict` to make that thappen, but, I could of course be wrong.

Reply:
in particular,
```diff
--- a/src/nvim/option.c
+++ b/src/nvim/option.c
@@ -4870,7 +4870,7 @@ int get_option_value_strict(char *name, int64_t *numval, char **stringval, int o
         // only getting a pointer, no need to use aucmd_prepbuf()
         curbuf = (buf_T *)from;
         curwin->w_buffer = curbuf;
-        varp = get_varp(p);
+        varp = get_varp_scope(p, OPT_LOCAL);
         curbuf = save_curbuf;
         curwin->w_buffer = curbuf;
       }
@@ -4878,7 +4878,7 @@ int get_option_value_strict(char *name, int64_t *numval, char **stringval, int o
       win_T *save_curwin = curwin;
       curwin = (win_T *)from;
       curbuf = curwin->w_buffer;
-      varp = get_varp(p);
+      varp = get_varp_scope(p, OPT_LOCAL);
       curwin = save_curwin;
       curbuf = curwin->w_buffer;
     }

```

Reply:
Yep I actually tried that in https://github.com/neovim/neovim/commit/e350feb13bdc8cd9ac941e7284ffc6f2021eeaec but IIRC it broke some other things so I reverted it.

> I just see no reason to implement some code/behaviour we don't want before the code we actually want.

I'll look into this deeper and figure out the right solution.

Reply:
>  it broke some other things so I reverted it.

What things did it break?

Reply:
Nothing insurmountable, it just caused some test failures: https://github.com/neovim/neovim/runs/3556434065

I didn't look into it too deeply at the time, I'll take another look when I get a chance.

Reply:
Well, it _is_ a change of behavior, so test failures are expected to be seen, and not an indication of the change being "wrong" (by necessity)